From cd8f057d5aed0cab86b8a8ddac037a6e94adec2d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Fri, 11 May 2012 10:31:16 +0300
Subject: [PATCH 3/5] Turn developer build checking off

The comment states that developer builds are detected by presence of
'private_tests' token in QT_CONFIG. That flag is set by QtBase build
when the option "-developer-build" is used.

QtWebKit build assumes that such a build is always local and that
certain paths are writable. Since we build Qt in modular fashion, this
assumption is false. Turn off the check, and thus skip the extra steps
where files are forcefully copied to system paths.
---
 Source/api.pri |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Source/api.pri b/Source/api.pri
index 8aa88e0..c434af0 100644
--- a/Source/api.pri
+++ b/Source/api.pri
@@ -44,7 +44,11 @@ haveQt(5) {
 
     # We use private_tests to detect developer build, since the destdir will
     # always be our webkit build dir. This might change as configure changes.
-    contains(QT_CONFIG, private_tests): CONFIG += qt_developer_build
+    # XXX: When building QtBase with "developer-build" flag, the
+    # privates tests are turned on. However, that flag also controls the
+    # assumptions here, so we comment out the following line
+    #
+    #contains(QT_CONFIG, private_tests): CONFIG += qt_developer_build
 
     !build_pass {
         # Make a more accessible copy of the module forward file
-- 
1.7.10

